### ISSUE TITLE
New-SafeguardAsset doesn't discover the host key

### ISSUE THREAD

**Issue by bnicholesdell** [OneIdentity/safeguard-ps#169]
The following command should create an asset and discover the host key:

    $newAsset = New-SafeguardAsset -NetworkAddress bnichvm1.sg.lab -Platform 68 -AcceptSshHostKey -Description "Linux Server*" -DisplayName "Linux Server" -ServiceAccountCredentialType None

The desktop UI currently allows a user to discover the host key without having to provide a service account.  NoSshHostKeyDiscovery was not specified and since the default value is false for this parameter, the commandlet should have tried to discover the host key.  The documentation for the commandlet says:

  -NoSshHostKeyDiscovery [<SwitchParameter>]
        Whether or not to skip SSH host key discovery for platforms that support it.

        Required?                    false
        Position?                    named
        Default value                False
        Accept pipeline input?       false
        Accept wildcard characters?  false

The output after getting the new asset.  The host key and thumbprint are empty:

Id                              : 11
Description                     : Linux Server*
PlatformId                      : 68
PlatformDisplayName             : SUSE Linux Enterprise Server (SLES) 12 x86_64
IsDirectory                     : False
ManagedNetworkId                :
ManagedNetworkName              :
CreatedDate                     : 2019-08-12T21:24:09Z
CreatedByUserId                 : 1
CreatedByUserDisplayName        : Brad Nicholes
LastAccountDiscoveryDate        :
LastSuccessAccountDiscoveryDate :
LastFailureAccountDiscoveryDate :
FailedAccountDiscoveryAttempts  : 0
NextAccountDiscoveryDate        :
LastServiceDiscoveryDate        :
LastSuccessServiceDiscoveryDate :
LastFailureServiceDiscoveryDate :
FailedServiceDiscoveryAttempts  : 0
NextServiceDiscoveryDate        :
Name                            : Linux Server
NetworkAddress                  : bnichvm1.sg.lab
AssetPartitionId                : -1
AssetPartitionName              : Macrocosm
LicenseClass                    : System
DirectoryProperties             : @{DirectoryId=; DirectoryName=; DomainName=; NetbiosName=; DistinguishedName=; ObjectGuid=;
                                  ObjectSid=}
DirectoryAssetProperties        : @{ForestRootDomain=; DomainName=; AllowSharedSearch=False; SynchronizationIntervalMinutes=;
                                  LastSynchronizedDate=; NextSynchronizedDate=; DeleteSyncIntervalMinutes=; LastDeleteSyncDate=;
                                  NextDeleteSyncDate=; Domains=System.Object[]; LastSuccessSynchronizedDate=;
                                  LastFailureSynchronizedDate=; LastSuccessDeleteSyncDate=; LastFailureDeleteSyncDate=;
                                  SchemaProperties=}
SessionAccessProperties         : @{SshSessionPort=22; RemoteDesktopSessionPort=; TelnetSessionPort=23}
Platform                        : @{Id=68; PlatformType=Suse; PlatformFamily=Linux; DisplayName=SUSE Linux Enterprise Server
                                  (SLES) 12 x86_64; Name=SUSE Linux Enterprise Server (SLES); Description=; Version=12;
                                  Architecture=x86_64; DeviceClass=Server; IsAcctNameCaseSensitive=True; LicenseClass=System;
                                  IsSystemOwned=True; SupportedOperations=System.Object[]; ConnectionProperties=;
                                  PasswordFeatureProperties=; SessionFeatureProperties=; CustomScriptProperties=}
Owners                          : {}
Tags                            : {}
ProfileId                       :
ProfileName                     :
EffectiveProfileId              : -1
EffectiveProfileName            : Macrocosm Profile
Disabled                        : False
AssetDiscoveryJobId             :
AssetDiscoveryJobName           :
AccountDiscoveryScheduleId      :
AccountDiscoveryScheduleName    :
CustomScriptParameters          : {}
ConnectionProperties            : @{ServiceAccountId=; ServiceAccountName=; ServiceAccountDomainName=;
                                  EffectiveServiceAccountName=; ServiceAccountPassword=; ServiceAccountHasPassword=False;
                                  Port=22; ServiceAccountSshKeyId=; ServiceAccountSshKeyFingerprint=;
                                  ServiceAccountSshKeyComment=; ServiceAccountAssetId=; ServiceAccountAssetName=;
                                  ServiceAccountAssetPlatformId=0; ServiceAccountAssetPlatformType=Unknown;
                                  ServiceAccountAssetPlatformDisplayName=; ServiceAccountNetbiosName=;
                                  ServiceAccountDistinguishedName=; ServiceAccountUniqueObjectId=; ServiceAccountSecurityId=;
                                  EffectiveServiceAccountDistinguishedName=; EnablePassword=; EnableHasPassword=False;
                                  CommandTimeout=20; WorkstationId=; ClientId=; ServiceAccountCredentialType=None;
                                  ServiceAccountProfileId=; ServiceAccountProfileName=; UseSslEncryption=False;
                                  VerifySslCertificate=False; Instance=; SslThumbprint=; PrivilegeElevationCommand=;
                                  AccessKeyId=; SecretKey=; HasSecretKey=False; OraclePrivileges=;
                                  UseNamedPipeForServiceAccountConnection=False}
AllowSessionRequests            : True
**SshHostKey                      :
SshHostKeyFingerprint           :**

**Comment by Goelrachit** [comment]
@DanPeterson explicitly added code to not discover ssh host key when no service account is provided. https://github.com/OneIdentity/safeguard-ps/commit/38e78612f74bd0ebf9eca76498de06d9aeee2120

I verified that we are able to discover the key if we remove the restriction. Is there a reason we want to keep this restriction? Maybe the capability to discover the key without service account was added after Dan added the code?
